test.py

```diff
@@ -681,11 +681,12 @@ def set_summary(summary):
                 set_summary("failed: {}".format(e))
             finally:
                 if self.is_new or self.is_equal_result is False:
-                    # Put a copy of the .reject file close to the .result file
-                    # so that it's easy to analyze the diff or overwrite .result
-                    # with .reject. Preserve the original .reject file: in
-                    # multiple modes the copy .reject file may be overwritten.
-                    shutil.copyfile(self.tmpfile, self.reject)
+                    if os.path.exists(self.tmpfile):
+                        # Put a copy of the .reject file close to the .result file
+                        # so that it's easy to analyze the diff or overwrite .result
+                        # with .reject. Preserve the original .reject file: in
+                        # multiple modes the copy .reject file may be overwritten.
+                        shutil.copyfile(self.tmpfile, self.reject)
```

Comment:
we need to unlink the file after copying. as the following branch does. The question is why we enter this branch. I'd protect this branch with self.is_executed_ok instead.

Reply:
> we need to unlink the file after copying. as the following branch does

So it wasn't unconditional, it would only unlink in case we decided not to copy.
Why would we want to now change the logic to unlink always? The commit explains why we want to preserve the original. @kostja

Reply:
I think not unlinking was another unrelated issue.

Reply:
Please provide evidence that there was an issue.

If there was no issue, don't do this change, continue copying.

If there was an issue, then:
1. explain the motivation of replacing copy with move in the commit message
2. update the comment as well

Reply:
You can also decide not to do this change at all in this PR (even if there's an issue), but if you decide to do it, do it all the way - including updating the comments and explaining the motivation in the commit message.